src/parser/block-additions/core-block.php

```diff
@@ -0,0 +1,218 @@
+		add_filter( 'vip_block_data_api__sourced_block_result', [ __CLASS__, 'remove_content_array' ], 5, 2 );
+	}
+
+	/**
+	 * Setup before render.
+	 */
+	public static function setup_before_render(): void {
+		/**
+		 * Hook into the `render_block` filter, which is near the end of WP_Block#render().
+		 * This allows us to capture the inner blocks of synced patterns ("core/block").
+		 * See `capture_inner_blocks`.
+		 */
+		add_filter( 'render_block', [ __CLASS__, 'capture_inner_blocks' ], 10, 3 );
+	}
+
+	/**
+	 * Cleanup after render.
+	 */
+	public static function cleanup_after_render() {
+		self::$captured_inner_blocks = [];
+		remove_filter( 'render_block', [ __CLASS__, 'capture_inner_blocks' ], 10 );
+	}
+
+	/**
+	 * Capture the inner blocks of synced patterns during block rendering. Intended
+	 * for use with the `render_block` filter.
+	 *
+	 * We have no intention of filtering the rendered block content, but this hook
+	 * is conveniently located near the end of WP_Block#render() after block
+	 * processing is finished. We get access to the parent block via the global
+	 * static class `WP_Block_Supports`.
+	 *
+	 * This approach is necessary because synced patterns (core/block) are dynamic
+	 * blocks, and core's method of rendering dynamic blocks severs the connection
+	 * between the parent block and its inner blocks:
+	 *
+	 * https://github.com/WordPress/WordPress/blob/6.6.1/wp-includes/class-wp-block.php#L519
+	 *
+	 * A dynamic block's render callback function returns only an HTML string.
+	 * This is suitable for most dynamic blocks, but synced patterns are just
+	 * references to other blocks, so it can be frustrating to see their contents
+	 * missing from the Block Data API. Capturing synced pattern content as inner
+	 * blocks is extremely useful and avoids the need for additional API calls.
+	 *
+	 * @param string   $block_content Rendered block content.
+	 * @param array    $parsed_block  Parsed block data.
+	 * @param WP_Block $block         Block instance.
+	 * @return string
+	 */
+	public static function capture_inner_blocks( string $block_content, array $parsed_block, WP_Block $block ): string {
+		// If this block is a synced pattern, that means it is finished rendering.
+		// Lock its inner blocks to prevent further captures in case it is rendered
+		// elsewhere in the tree.
+		if ( self::$block_name === $block->name ) {
+			$store_key = self::get_store_key( $parsed_block );
+			if ( isset( self::$captured_inner_blocks[ $store_key ] ) ) {
+				self::$captured_inner_blocks[ $store_key ]['locked'] = true;
+			}
+		}
+
+		// Get the parent block that is currently being rendered. This is fragile,
+		// but is currently the only way we can get access to the parent block from
+		// inside a dynamic block's render callback function.
+		//
+		// https://github.com/WordPress/WordPress/blob/6.6.1/wp-includes/class-wp-block.php#L517
+		$parent_block = WP_Block_Supports::$block_to_render;
```

Comment:
Can we wrap this in some sort of safety method in case it becomes private or changes names in the future? I think we can just bail if we don't see the variable we expect, and we'll see that synced pattern tests are failing.